### PR TITLE
Add UT for DirectoriesStrategy

### PR DESCRIPTION
**Pull request by mdf369**
Add unit test for all strategies of choosing directories for new file.

### CODE REVIEW

File: iotdb/src/main/java/org/apache/iotdb/db/conf/directories/strategy/SequenceStrategy.java [hunk 1]
```diff
@@ -53,7 +49,7 @@ public int nextFolderIndex() throws DiskSpaceInsufficientException {
 
   private int tryGetNextIndex(int start) throws DiskSpaceInsufficientException {
     int index = (start + 1) % folders.size();
-    while (!hasSpace(folders.get(index))) {
+    while (!FileUtils.hasSpace(folders.get(index))) {
```
Comment (jixuan1989) [inline iotdb/src/main/java/org/apache/iotdb/db/conf/directories/strategy/SequenceStrategy.java]: suppose there is no space any more, then start == -1, and then,.... the process will never finish the while-loop...

File: iotdb/src/main/java/org/apache/iotdb/db/conf/directories/strategy/MaxDiskUsableSpaceFirstStrategy.java [hunk 2]
```diff
@@ -35,7 +36,12 @@ public int getMaxSpaceFolder() throws DiskSpaceInsufficientException {
...
 
     for (int i = 0; i < folders.size(); i++) {
-      long space = getUsableSpace(folders.get(i));
+      String folder = folders.get(i);
+      if (!FileUtils.hasSpace(folder)) {
+        continue;
+      }
+
+      long space = FileUtils.getUsableSpace(folder);
       if (space > maxSpace) {
```
Comment (jixuan1989) [inline iotdb/src/main/java/org/apache/iotdb/db/conf/directories/strategy/MaxDiskUsableSpaceFirstStrategy.java]: should be "if( space >= maxSpace)... "
Suppose all folder are empty, if so, you will return -1...

File: iotdb/src/main/java/org/apache/iotdb/db/conf/directories/strategy/SequenceStrategy.java (hunk 1 above)
Comment (jixuan1989) [reply iotdb/src/main/java/org/apache/iotdb/db/conf/directories/strategy/SequenceStrategy.java]: Ok... seems that you guarantee that there must be available space by super.init(). If so, please add comments in the init().

Comment (mdf369) [reply iotdb/src/main/java/org/apache/iotdb/db/conf/directories/strategy/SequenceStrategy.java]: added.

File: iotdb/src/main/java/org/apache/iotdb/db/conf/directories/strategy/MaxDiskUsableSpaceFirstStrategy.java (hunk 2 above)
Comment (mdf369) [reply iotdb/src/main/java/org/apache/iotdb/db/conf/directories/strategy/MaxDiskUsableSpaceFirstStrategy.java]: if all folders are empty, maxIndex will not be init as line 41 skip to next loop, so exception will be thrown.

**Pull request merged**